### PR TITLE
Adds resubmit_facts config for puppet.conf

### PR DESCRIPTION
**Pull request by crazymind1337**
This adds the possibility to configure `resubmit_facts` via a parameter.

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -3,37 +3,38 @@
...
-  $prerun_command      = undef
-  $postrun_command     = undef
-  $server_compile_mode = undef
-  $dns_alt_names       = []
-  $use_srv_records     = false
+  $cron_cmd             = undef
+  $systemd_cmd          = undef
+
+  $agent_noop           = false
+  $resubmit_facts = false
```
Comment (ekohl) [inline manifests/params.pp:24]: Please keep this aligned

File: manifests/params.pp [hunk 2]
```diff
@@ -3,37 +3,38 @@
...
-  $ca_server           = undef
-  $ca_port             = undef
-  $ca_crl_filepath     = undef
-  $server_crl_enable   = undef
-  $prerun_command      = undef
-  $postrun_command     = undef
-  $server_compile_mode = undef
-  $dns_alt_names       = []
-  $use_srv_records     = false
+  $cron_cmd             = undef
```
Comment (ekohl) [inline manifests/params.pp:20]: There's no need to realign

File: manifests/params.pp (hunk 1 above)
Comment (wbclark) [inline manifests/params.pp:24]: Thanks for opening this PR @crazymind1337 !

Could you fix the alignment for the new parameter here please?

Also, I don't see any reason why the alignment for the other params needs to change.

File: manifests/init.pp [hunk 3]
```diff
@@ -600,6 +600,7 @@
   Hash[String, Data] $additional_settings = $puppet::params::additional_settings,
   Hash[String, Data] $agent_additional_settings = $puppet::params::agent_additional_settings,
   Optional[String] $agent_restart_command = $puppet::params::agent_restart_command,
+  Boolean $resubmit_facts = $puppet::params::resubmit_facts,
```
Comment (ekohl) [inline manifests/init.pp:603]: Would need to be documented

**Review by ekohl — CHANGES_REQUESTED** [review]
I wonder if this is really such a common thing. We try to only have all the common parameters in `init.pp`. Are you not able to keep this in your own manifest?

**Comment by crazymind1337** [comment]
Wow that has been fast :D I will fix the mentioned issues.

File: manifests/params.pp (hunk 1 above)
Comment (crazymind1337) [reply manifests/params.pp:24]: I renamed the variable from previous `$agent_resubmit_facts` via `sed` and forgot to fix the alignment :(

Comment (crazymind1337) [reply manifests/params.pp:24]: I renamed the variable from previous `$agent_resubmit_facts` via `sed` and forgot to fix the alignment :(

File: manifests/params.pp (hunk 2 above)
Comment (crazymind1337) [reply manifests/params.pp:20]: I renamed the variable from previous `$agent_resubmit_facts` via `sed` and forgot to fix the alignment :(

File: manifests/params.pp (hunk 1 above)
Comment (crazymind1337) [reply manifests/params.pp]: Fixed.

File: manifests/params.pp (hunk 2 above)
Comment (crazymind1337) [reply manifests/params.pp]: Fixed.

File: manifests/init.pp (hunk 3 above)
Comment (crazymind1337) [reply manifests/init.pp:605]: Fixed.

File: manifests/params.pp (hunk 1 above)
Comment (crazymind1337) [reply manifests/params.pp]: Fixed.

**Comment by crazymind1337** [comment]
Do you still need some changes?

**Comment by ekohl** [comment]
It's been busy with the Foreman 2.3.0 RC1 preparations, so a bit delayed on the response.

> I wonder if this is really such a common thing. We try to only have all the common parameters in `init.pp`. Are you not able to keep this in your own manifest?

I'd still like to get this question answered.

**Comment by crazymind1337** [comment]
> I wonder if this is really such a common thing. We try to only have all the common parameters in `init.pp`. Are you not able to keep this in your own manifest?

It is a new feature for since puppet 6.11.0 (?). Yes it also could be done in our manifests.

**Comment by ekohl** [comment]
> Yes it also could be done in our manifests.

I think this use case isn't that common and I think doing it in your manifests makes more sense.

**Comment by crazymind1337** [comment]
Ok

**Pull request closed without merging**